homeassistant/components/mqtt/notify.py

```diff
@@ -0,0 +1,393 @@
+            """Handle discovery update."""
+            async_dispatcher_send(
+                hass, MQTT_DISCOVERY_DONE.format(service.discovery_hash), None
+            )
+            if not discovery_payload:
+                # unregister notify service through auto discovery
+                await async_tear_down_service()
+                return
+
+            # update notify service through auto discovery
+            await service.async_update_service(discovery_payload)
+            _LOGGER.debug(
+                "Notify service %s updated has been processed",
+                service.discovery_hash,
+            )
+
+        async def async_device_removed(event):
+            """Handle the removal of a device."""
+            device_id = event.data["device_id"]
+            if (
+                event.data["action"] != "remove"
+                or device_id != service.device_id
+                or self._device_removed
+            ):
+                return
+            self._device_removed = True
+            await async_tear_down_service()
+
+        async def async_tear_down_service():
+            """Handle the removal of the service."""
+            services = hass.data.setdefault(MQTT_NOTIFY_SERVICES_SETUP, {})
+            if self._service.service_name in services.keys():
+                del services[self._service.service_name]
+            if not self._device_removed and service.device_id:
+                self._device_removed = True
+                await cleanup_device_registry(hass, service.device_id)
+            clear_discovery_hash(hass, service.discovery_hash)
+            self._remove_discovery()
+            await service.async_unregister_services()
+            _LOGGER.info(
+                "Notify service %s has been removed",
+                service.discovery_hash,
+            )
+            del self._service
+
+        self._service = service
+        self._remove_discovery = async_dispatcher_connect(
+            hass,
+            MQTT_DISCOVERY_UPDATED.format(service.discovery_hash),
+            async_discovery_update,
+        )
+        if service.device_id:
+            self._remove_device_updated = hass.bus.async_listen(
+                EVENT_DEVICE_REGISTRY_UPDATED, async_device_removed
+            )
+        self._device_removed = False
+        async_dispatcher_send(
+            hass, MQTT_DISCOVERY_DONE.format(service.discovery_hash), None
+        )
+        _LOGGER.info(
+            "Notify service %s has been initialized",
+            service.discovery_hash,
+        )
+
+
+class MqttNotificationService(notify.BaseNotificationService):
+    """Implement the notification service for MQTT."""
+
+    def __init__(
+        self,
+        hass: HomeAssistant,
+        command_topic: str,
+        command_template: MqttCommandTemplate,
+        encoding: str,
+        name: str | None,
+        qos: int,
+        retain: bool,
+        targets: list,
+        title: str | None,
```

Comment:
The verbosity is nice for typing, but it adds quite a lot of lines of code since we handle the config in several places, and then pull out keys and store them as instance variables.
Could we instead pass in the config as a typed dict, since we know what's there after config validation?

Reply:
Solved this, please check the type supressing